sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3275,16 +3312,41 @@ pi_result piDevicePartition(pi_device Device,
     return Res;
   }
 
+  auto EffectiveSize = [&]() -> decltype(Device->SubDevices.size()) {
```

Comment:
```suggestion
  auto NumDevices = [&]() -> decltype(Device->SubDevices.size()) {
```

Reply:
I'll change to `EffectiveNumDevices` as I think it's important to note the difference between HW/logical partitioning in this area of the code.